public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, DataSourceSettings, SelectableValue } from '@grafana/data';
+import { DataSourceHttpSettings, FormLabel, Input, SecretFormField, Select } from '@grafana/ui';
+import { InfluxOptions, InfluxSecureJsonData } from '../types';
+
+export type Props = DataSourcePluginOptionsEditorProps<InfluxOptions>;
+
+type InfluxDataSourceSettings = DataSourceSettings<InfluxOptions, InfluxSecureJsonData>;
+
+export interface State {
+  config: InfluxDataSourceSettings;
```

Comment:
In general, it looks like putting settings in the state complicates all the logic because you then need to worry about secure fields explicilty.  Can we try using props directly?

Reply:
Well, if I use it directly from props, it may not have all of the object keys on it that are needed.

Reply:
for example?  

defaults?  if so, you can assign defaults at the beginning of render(), no?

Reply:
Why is this still needed?